src/dagmc/dagmcmetadata.hpp

```diff
@@ -8,118 +8,316 @@
 
 class dagmcMetaData {
```

Comment:
This class would probably also benefit from some top level general description of how it all works, e.g. what's a "property" and how are key:value pairs specified, etc...

Reply:
Gave it a go. Lemme know what you think.